service/history/visibilityQueueTaskExecutor.go

```diff
@@ -73,18 +74,25 @@ func newVisibilityQueueTaskExecutor(
 func (t *visibilityQueueTaskExecutor) Execute(
 	ctx context.Context,
 	executable queues.Executable,
-) error {
-	switch task := executable.GetTask().(type) {
+) (metrics.Scope, error) {
+
+	task := executable.GetTask()
+	scope := t.shard.GetMetricsClient().Scope(
+		tasks.GetVisibilityTaskMetricsScope(task),
+		getNamespaceTagByID(t.shard.GetNamespaceRegistry(), task.GetNamespaceID()),
+	)
```

Comment:
> some metrics that are across multiple executions

multiple attempt? 
if so the task should also track this.

Reply:
Yes multiple attempts. Those information is currently tracked by task `executable`. With existing implementation, I don't think `executor` should track those as it only contains the execution logic for one attempt. Scope is returned here because task executable, who emits all the metrics today, needs to know if the task is processed as active or passive and use different metric scopes.

I think we've discussed splitting existing task executor and make each task type has its own executor. In that world, probably there's no difference between executable and executor, and we don't need to worry about this question.